.github/workflows/gitguardian.yml

```
@@ -0,0 +1,96 @@
+# Configure:
+#   Repository secret: GITGUARDIAN_API_KEY (service account or PAT with scan scope).
+# Optional Telegram on scan failure only (TELEGRAM_BOT_TOKEN + TELEGRAM_CHAT_ID).
+#
+# If the API key is unset, the workflow succeeds without scanning (same pattern as SonarCloud gate).
+
+name: GitGuardian
+
+on:
+  push:
+    branches:
+      - ${{ vars.PRIMARY_BRANCH || 'main' }}
```

actionlint (1.7.12)

[error] character ' ' is invalid for branch and tag names. ref name cannot contain spaces, ~, ^, :, [, ?, *. see `man git-check-ref-format` for more details. note that regular expression is unavailable. note: filter pattern syntax is explained at https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

(glob)